### PR TITLE
[duplicate] Automatically enable depends of mods when enabling the mods in the world config menu

### PR DESCRIPTION
**Pull request by HybridDog**
l can't reopen this, so l did a new request: #3473

### CODE REVIEW

File: builtin/mainmenu/dlg_config_world.lua [hunk 1]
```diff
@@ -54,7 +43,8 @@ local function get_formspec(data)
 		"button[9.25,6.35;2,0.5;btn_config_world_save;" .. fgettext("Save") .. "]" ..
 		"button[7.4,6.35;2,0.5;btn_config_world_cancel;" .. fgettext("Cancel") .. "]"
 
-	if mod ~= nil and mod.name ~= "" and mod.typ ~= "game_mod" then
+	if mod.name ~= ""
+	and mod.typ ~= "game_mod" then
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]: 2 intends here or put it on the same line (it's short enough).

File: builtin/mainmenu/dlg_config_world.lua [hunk 2]
```diff
@@ -91,30 +81,95 @@ local function get_formspec(data)
...
 end
 
+-- makes a list of mod ids indexed by their names
+local mod_ids
+local function get_mod_ids(list)
+	mod_ids = {}
+	for id,mod in pairs(list) do
+		if mod.typ == "global_mod"
+		and not mod.is_modpack
+		and mod.name then
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]: Same here.

File: builtin/mainmenu/dlg_config_world.lua [hunk 3]
```diff
@@ -91,30 +81,95 @@ local function get_formspec(data)
...
+		get_mod_ids(list)
+	end
+
+	local to_enable = {}
+	for name in pairs(enabled_mods) do
+		local depends = modmgr.get_dependencies(list[mod_ids[name]].path, true)
+		if depends then
+			for _,name in pairs(depends) do
+				if not enabled_mods[name] then
+					table.insert(to_enable, name)
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]: Use `to_enable[#to_enable+1] = name` (faster).

File: builtin/mainmenu/dlg_config_world.lua [hunk 4]
```diff
@@ -148,14 +203,14 @@ local function handle_buttons(this, fields)
 		return true
 	end
 
-	if fields["cb_hide_gamemods"] ~= nil or
-		fields["cb_hide_mpcontent"] ~= nil then
+	if fields["cb_hide_gamemods"] ~= nil
+	or fields["cb_hide_mpcontent"] ~= nil then
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]: 2 intends.

File: builtin/mainmenu/dlg_config_world.lua [hunk 5]
```diff
@@ -91,30 +81,95 @@ local function get_formspec(data)
...
+			local id = mod_ids[name]
+			if id then
+				local mod = list[id]
+				mod.enabled = true
+				enabled_mods[name] = true
+				local depends = modmgr.get_dependencies(mod.path, true)
+				if depends then
+					for _,name in pairs(depends) do
+						if not enabled_mods[name] then
+							table.insert(to_enable, name)
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua:159]: ^

**Comment by HybridDog** [comment]
l can't reopen this, so l did a new request: #3473

**Pull request closed without merging**